common/models/model-definition.js

```diff
@@ -75,10 +75,19 @@ function ready(ModelDefinition) {
     assert(relation.embed && relation.embed.as, 'embed requires "as"');
     switch (relation.embed.as) {
       case 'object':
-        assert(relation.embed.key, 'embed as object requires "key"');
+        assert(relation.embed.key || relation.embed.keySetter,
+          'embed as object requires "key" or "keySetter"');
         result = {};
         relatedData.forEach(function(related) {
-          var key = related[relation.embed.key];
+          var Definition = app.models[relation.model];
+          var key;
+          if (relation.embed.key) {
+            key = related[relation.embed.key];
+          }
+          var keySetter = relation.embed.keySetter;
```

Comment:
Yeah, I think we are in agreement, just the naming convention is different. We should have a `keyGetter` and a `keySetter`.
`key` will become `keySetter` and `keyFn` will be `keyGetter`